Main refactoring of v6.3 diffraction release notes

sf1919 opened the pull request:
**Description of work.**
The main refactoring the the Diffraction release notes ahead of release of 6.3. Includes:-

- Removing the developer warning text
- Adding extra subheadings
- Reordering  and grouping of notes where appropriate
- Rewording notes to improve consistency
- Adding links
- Spelling & Grammar check

Further changes will be made later when any separate files for v6.3 diffraction release notes are amalgamated into the final release notes.

**To test:**

- remove build/docs/doctrees
- build the documents again and check there are no warnings
- Check that the links work
- Check the information is still correct
- Grammar/Spelling check


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

sf1919 committed:
Moved release note from 6.2 to 6.3

A release note was accidentally added to 6.2 notes despite referring to a change available in 6.3

sf1919 committed:
Updated Poweder Diffraction 6.3 release notes

- Rearranged unorganised release notes
- Formatted notes
- Grouped notes where appropriate
- Organised release notes in each section alphabetically by algorithm where possible
- Some slight rewording for consistency and readability
- Addition of Deprecation section

sf1919 committed:
Updated Engineering Diffraction v6.3 release notes

- Adding some formatting
- Grouped speed improvements and Fitting Tab improvements
- Added links where appropriate
- Minor rewording for consistency/readability

sf1919 committed:
Updated Single Crystal Diffraction Release notes for v6.3

- Added New features section and rearranged notes accordingly
- Minor wording changes for readability
- Minor formatting

sf1919 committed:
Additional minor formatting changes

DanielMurphy22 commented on docs/source/release/v6.3.0/diffraction.rst:
```diff
@@ -5,78 +5,96 @@ Diffraction Changes
 - The integration range has been corrected inside :ref:`PDFFourierTransform v2 <algm-PDFFourierTransform-v2>`.
-- Fix problem with the create_vanadium action when running with tt_mode=Custom in the ISIS PEARL powder diffraction scripts. Create a separate Vanadium file for each different custom grouping file rather than one for all custom runs
+- :ref:`SaveFocusedXYE <algm-SaveFocusedXYE>` now correctly writes all spectra to a single file when ``SplitFiles`` is ``False``. Previously it wrote only a single spectrum.
+- For processing vanadium run it no longer finds the environment automatically in :ref:`SetSampleFromLogs <algm-SetSampleFromLogs>`.
```
Not really sure how the PR https://github.com/mantidproject/mantid/pull/32548 relates to this release note! 
```suggestion
- :ref:`SetSampleFromLogs <algm-SetSampleFromLogs>` no longer finds the environment automatically, when processing vanadium runs .
```
Does that sound okay @peterfpeterson ?

sf1919 replied:
I've checked and this came from PR #32537 . @Kvieta1990 can you help with this?

peterfpeterson replied:
How about
```rst
Add option to enable (default on) finding the sample environment automatically using :ref:`SetSampleFromLogs <algm-SetSampleFromLogs>`. This is used to turn off the feature for vanadium measurements when using :ref:`mantid.utils.absorptioncorrutils`
```

DanielMurphy22 commented on docs/source/release/v6.3.0/diffraction.rst:
```diff
@@ -5,78 +5,96 @@ Diffraction Changes
+  * The tab has been made more tolerant to users deleting or renaming the workspaces in the workbench Workspaces widget.
+
+* Updated the default values for :ref:`EnggEstimateFocussedBackground <algm-EnggEstimateFocussedBackground>` and in the fitting tab table to ``Niter = 50`` and ``XWindow = { 600 for TOF, 0.02 for d-Spacing }``.
+* The file filter in the Focus tab for calibration Region includes ``No Region Filter``, ``North``, ``South`` and now also ``Cropped``, ``Custom``, ``Texture`` and ``Both Banks``. The text for ``No Unit/Region Filter`` are colored grey.
```
```suggestion
* The file filter in the Focus tab for calibration Region includes ``No Region Filter``, ``North``, ``South`` and now also ``Cropped``, ``Custom``, ``Texture`` and ``Both Banks``. The text for ``No Unit/Region Filter`` is colored grey.
```
This looks like my PR! Sorry!

DanielMurphy22 commented on docs/source/release/v6.3.0/diffraction.rst:
```diff
@@ -5,78 +5,96 @@ Diffraction Changes
 New features
 ############
-- Now support two texture grouping schemes: Texture20 (10 groups per bank, 20 in total) and Texture30 (15 groups per bank, 30 in total) for ENGIN-X in the :ref:`Engineering Diffraction interface<Engineering_Diffraction-ref>`. Note this involved changes to the bankID log values saved with focused data, so this means the UI will not load in previously focused .nxs files.
+- Now supports two texture grouping schemes: ``Texture20`` (10 groups per bank, 20 in total) and ``Texture30`` (15 groups per bank, 30 in total) for ``ENGIN-X`` in the :ref:`Engineering Diffraction interface<Engineering_Diffraction-ref>`. Note this involved changes to the ``bankID`` log values saved with focused data, so this means the UI will not load in previously focused ``.nxs`` files.
```
This looks like the most impressive thing on this page! Here's an image from @RichardWaiteSTFC for this note
![TextureGroup20](https://user-images.githubusercontent.com/55980573/148940797-7a87deb2-0e8f-41a0-ab13-834583ab6032.png)
.

DanielMurphy22 reviewed (changes requested):
Super good job! Just a few nitpicks and an image!

sf1919 committed:
Revisions after review

- added image for new texture grouping schemes in Engineering Diffraction
- Rewording of a couple of release notes for clarity

DanielMurphy22 reviewed (approved):
Thank you for the effort in particular for the Diffraction release notes! They look really good!
